gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/remote/EmbeddedRemoteConnection.java

```diff
@@ -59,7 +62,11 @@ public <E> CompletableFuture<RemoteTraversal<?, E>> submitAsync(final GremlinLan
         // the new submit() in 3.3.x when the deprecation is removed
         final CompletableFuture<RemoteTraversal<?, E>> promise = new CompletableFuture<>();
         try {
-            promise.complete(new EmbeddedRemoteTraversal(JavaTranslator.of(g).translate(gremlinLang)));
+            final GremlinLangScriptEngine scriptEngine = new GremlinLangScriptEngine();
```

Comment:
do we need a new instance each time here? i assume this could be created per instance?